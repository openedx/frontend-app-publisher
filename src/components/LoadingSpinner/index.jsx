import React from 'react';
import { Spinner } from '@edx/paragon';
import PropTypes from 'prop-types';

const LoadingSpinner = props => (
  <div className="mx-auto text-center">
    <div>
      { props.message }
    </div>
    <Spinner animation="border" className="mie-3" screenReaderText="loading" />
  </div>
);

LoadingSpinner.defaultProps = {
  message: 'Loadingâ€¦',
};

LoadingSpinner.propTypes = {
  message: PropTypes.string,
};

export default LoadingSpinner;
