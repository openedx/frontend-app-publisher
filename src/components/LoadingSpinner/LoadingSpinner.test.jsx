import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import LoadingSpinner from './index';

describe('LoadingSpinner', () => {
  it('shows a loading spinner', () => {
    const component = shallow(<LoadingSpinner />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
