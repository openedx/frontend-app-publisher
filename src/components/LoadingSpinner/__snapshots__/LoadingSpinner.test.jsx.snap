// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner shows a loading spinner 1`] = `
<div
  className="mx-auto text-center"
>
  <div>
    Loadingâ€¦
  </div>
  <ForwardRef
    animation="border"
    className="mie-3"
    screenReaderText="loading"
  />
</div>
`;
